zenpacklib.py

```diff
@@ -760,7 +760,7 @@ def setIdsInRelationship(self, relationship, ids):
         query = Or(*[Eq('id', x) for x in changed_ids])
 
         obj_map = {}
-        for result in self.device().search('ComponentBase', query):
+        for result in catalog_search(self.device(), 'ComponentBase', query):
             obj_map[result.id] = result.getObject()
```

Comment:
Looks like catalog_search handles cases where the device doesn't have a search method defined??